src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -128,7 +120,15 @@ public async Task<MetadataAsSourceFile> GetGeneratedFileAsync(Project project, I
                     {
                         try
                         {
-                            temporaryDocument = await DecompileSymbolAsync(temporaryDocument, symbol, cancellationToken).ConfigureAwait(false);
+                            var decompiledSourceService = temporaryDocument.Project.LanguageServices.GetService<IDecompiledSourceService>();
```

Comment:
there is a .GetLanguageService extension on Document that you can use to simplify this.